Minimum golang version needed

While try to start kcp by using go `v1.16.3` I got the following error:

```shell
# sigs.k8s.io/json/internal/golang/encoding/json
../../go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
../../go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```

AFAIU it's pretty much related to an older golang version so I updated to latest `v1.18.3` and everything started fine.

Maybe the README could report the minimum required golang version needed to run kcp without any troubles but I am not sure from which version the above was fixed. Not sure `v1.17.0` is already fine.

--- a/README.md
+++ b/README.md
@@ -151,6 +151,8 @@ However, `kcp` is not an acronym.
 
 ## How do I get started?
 
+First of all, be sure to have [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) and [Go](https://golang.org/doc/install) (1.17+) installed.
+
 After cloning the repository, you can start kcp on your machine using this command:
 
 ```

